helm-frontend/src/components/MiniLeaderboardTables.tsx

```diff
@@ -0,0 +1,303 @@
+    sortFirstMetric ? 1 : undefined,
+  );
+  const [activeGroupsTable, setActiveGroupsTable] = useState<GroupsTable>({
+    ...groupsTables[activeGroup],
+  });
+  const [sortDirection, setSortDirection] = useState<number>(1);
+  const [filteredModels, setFilteredModels] =
+    useState<string[]>(modelsToFilter);
+
+  interface HeaderValueObject {
+    value: string;
+  }
+
+  function truncateHeader(value: string): string {
+    if (value.length > 30) {
+      return value.substring(0, 27) + "...";
+    }
+    return value;
+  }
+
+  // TODO remove truncation once a visually suitable version of wrapping is determined
+  const getHeaderValue = (headerValueObject: HeaderValueObject): string => {
+    if (headerValueObject.value === "Model/adapter") {
+      return "Model";
+    } else if (headerValueObject.value.includes("-book")) {
+      return truncateHeader(headerValueObject.value.replace("-book", ""));
+    } else {
+      return truncateHeader(headerValueObject.value);
+    }
+  };
+
+  const [schema, setSchema] = useState<Schema | undefined>(undefined);
+
+  useEffect(() => {
+    const controller = new AbortController();
+    async function fetchData() {
+      const schema = await getSchema(controller.signal);
+      setSchema(schema);
+    }
+
+    void fetchData();
+    return () => controller.abort();
+  }, []);
+
+  const getModelDesc = (model: string): string => {
+    if (schema) {
+      const foundItem = schema.models.find(
+        (item) => item.display_name === model,
+      );
+      if (foundItem) {
+        let toRet = foundItem.description;
+        if (toRet.includes("/")) {
+          toRet = toRet.replace("/", "_");
+          return toRet;
+        } else {
+          return toRet;
+        }
+      }
+    }
+    return "";
+  };
+
+  const getModelForRunName = (model: string): string => {
+    if (schema) {
+      const foundItem = schema.models.find(
+        (item) => item.display_name === model,
+      );
+      if (foundItem) {
+        let toRet = foundItem.name;
+        if (toRet.includes("/")) {
+          toRet = toRet.replace("/", "_");
+          return toRet;
+        } else {
+          return toRet;
+        }
+      }
+    }
+    return "";
+  };
+  // create delimiter to parse out run group name (need to replace hyphen as some run names have hyphens in them)
+  /*function replaceLastHyphen(str: string): string {
+		const lastIndex = str.lastIndexOf(" - ");
+		if (lastIndex === -1) {
+			return str;
+		}
+		return str.substring(0, lastIndex) + "*" + str.substring(lastIndex + 1);
+	}
+	/*const getGroupForRunName = (rawGroup: string): string => {
+		const groupSplit = replaceLastHyphen(rawGroup).split("*");
+		const group = groupSplit[0].trim();
+		if (schema) {
+			const foundItem = schema.run_groups.find(
+				(item) =>
+					item.display_name === group || item.short_display_name === group
+			);
+			if (foundItem) {
+				return foundItem.name;
+			}
+		}
+		return "";
+	};*/
+
+  useEffect(() => {
```

Comment:
After cleaning up the state, all these `useEffect()`s should be removed as well; only loading the schema schema needs to be in a `useEffect()`.